boot/seal.go

```diff
@@ -54,13 +54,18 @@ var (
 // Hook functions setup by devicestate to support device-specific full
 // disk encryption implementations. The state must be locked when these
 // functions are called.
+//
+// XXX: move to "kernel/fde/fde.go" ?
 var (
 	HasFDESetupHook = func() (bool, error) {
 		return false, nil
 	}
 	RunFDESetupHook fde.RunSetupHookFunc = func(req *fde.SetupRequest) ([]byte, error) {
 		return nil, fmt.Errorf("internal error: RunFDESetupHook not set yet")
 	}
+	RunFDEDeviceSetupHook fde.RunDeviceSetupHookFunc = func(req *fde.SetupRequest) ([]byte, error) {
```

Comment:
This needs to be removed, it's better to even look at the op to determine locking than to have this extra hook.

Reply:
This is now removed and the lock is taken based on the "op" (not great but better than having two confusing hooks here). - Thanks to Samuele again.